platforms/software/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/SelfResolvingDependencyIntegrationTest.groovy

```diff
@@ -39,20 +39,25 @@ dependencies {
 task verify {
     doLast {
         def dep = configurations.compile.dependencies.find { it instanceof FileCollectionDependency }
-        println "files: " + dep.resolve().collect { it.name }
-        println "files-not-transitive: " + dep.resolve(false).collect { it.name }
-        println "files-transitive: " + dep.resolve(true).collect { it.name }
+        println "files: " + dep.files.files.collect { it.name }
+        println "resolve: " + dep.resolve().collect { it.name }
+        println "resolve-not-transitive: " + dep.resolve(false).collect { it.name }
+        println "resolve-transitive: " + dep.resolve(true).collect { it.name }
+        dep.getBuildDependencies()
     }
 }
 """
 
         when:
+        executer.expectDocumentedDeprecationWarning("Directly resolving a file collection dependency's files has been deprecated. This will fail with an error in Gradle 9.0. Add the dependency to a resolvable configuration and resolve the configuration. Consult the upgrading guide for further information: https://docs.gradle.org/current/userguide/upgrading_version_8.html#deprecate_self_resolving_dependency")
```

Comment:
🤔 This would be clearer splitting the deprecation message generating `resolve` into a separate test than the replacement.

Or maybe as a parameterized test, making clear to only expect the dep warnings using the older methods.

Reply:
Would rather leave this as-is to reduce duplication. We'll be deleting the deprecated parts of this test soon enough